api/src/main/java/org/apache/iceberg/actions/RewriteDataFiles.java

```diff
@@ -52,6 +52,13 @@ public interface RewriteDataFiles
 
   int PARTIAL_PROGRESS_MAX_COMMITS_DEFAULT = 10;
 
+  /**
```

Comment:
Not directly related to this change but I wonder whether we should throw an exception if all jobs fail. That would mean nothing was compacted. I think @RussellSpitzer had some ideas around this.

Thoughts, @manuzhang @RussellSpitzer @szehon-ho @amogh-jahagirdar @nastra?

Overall, it seems reasonable to me to have such a config. I'll check the implementation in detail later.

Reply:
@aokolnychyi do you have time to check in detail yet?

Reply:
I think the idea is reasonable, I still want to hear from @RussellSpitzer, as he mentioned some ideas before. Let me ping him offline.

Reply:
One thing we should probably do in a follow up is add this information to the output of the Procedure as well. So the procedure can report the percentage of commits that succeeded.

Reply:
I believe we already extended the action and procedure with the number of failed data files. We may check if the number of failed jobs should be propagated separately. If everyone is onboard, I'll take a look at this PR tomorrow.

Reply:
@aokolnychyi do you get a change to check this PR?